cisc/certificate.go

```diff
@@ -0,0 +1,321 @@
+c3Ryb290Y2F4My5wN2MwHwYDVR0jBBgwFoAUxKexpHsscfrb4UuQdf/EFWCFiRAw
+VAYDVR0gBE0wSzAIBgZngQwBAgEwPwYLKwYBBAGC3xMBAQEwMDAuBggrBgEFBQcC
+ARYiaHR0cDovL2Nwcy5yb290LXgxLmxldHNlbmNyeXB0Lm9yZzA8BgNVHR8ENTAz
+MDGgL6AthitodHRwOi8vY3JsLmlkZW50cnVzdC5jb20vRFNUUk9PVENBWDNDUkwu
+Y3JsMB0GA1UdDgQWBBSoSmpjBH3duubRObemRWXv86jsoTANBgkqhkiG9w0BAQsF
+AAOCAQEA3TPXEfNjWDjdGBX7CVW+dla5cEilaUcne8IkCJLxWh9KEik3JHRRHGJo
+uM2VcGfl96S8TihRzZvoroed6ti6WqEBmtzw3Wodatg+VyOeph4EYpr/1wXKtx8/
+wApIvJSwtmVi4MFU5aMqrSDE6ea73Mj2tcMyo5jMd6jmeWUHK8so/joWUoHOUgwu
+X4Po1QYz+3dszkDqMp4fklxBwXRsW10KXzPMTZ+sOPAveyxindmjkW8lGy+QsRlG
+PfZ+G6Z6h7mjem0Y+iWlkYcV4PIWL1iwBi8saCbGS5jN2p8M+X+Q7UNKEkROb3N6
+KOqkqm57TH2H3eDJAkSnh6/DNFu0Qg==
+-----END CERTIFICATE-----`
+
+func compare(cOld, cNew, k string) {
+	log.Info("Compare old against new cert: " + k)
+	cert1, err := pemToCertificate(cOld)
+	if err != nil {
+		log.Info("the old value is not a cert")
+		return
+	}
+	cert2, err := pemToCertificate(cNew)
+	if err != nil {
+		log.Info("the new value is not a cert")
+		return
+	}
+	if cert1.Equal(cert2) {
+		log.Info("No change in the certificate")
+
+	} else {
+		log.Info("The new certificate is different :")
+		if !((cert1.NotAfter).Equal(cert2.NotAfter)) {
+			log.Info("    Expiration date modification: " + cert1.NotAfter.String() + " to " + cert2.NotAfter.String())
+		}
+		if !((cert1.Subject.CommonName) == (cert2.Subject.CommonName)) {
+			log.Info("    Subject modification: " + cert1.Subject.CommonName + " to " + cert2.Subject.CommonName)
+		}
+		if !((cert1.Issuer.CommonName) == (cert2.Issuer.CommonName)) {
+			log.Info("    Issuer modification: " + cert1.Issuer.CommonName + " to " + cert2.Issuer.CommonName)
+		}
+
+		if !((cert1.PublicKey.(*rsa.PublicKey).N).Cmp(cert2.PublicKey.(*rsa.PublicKey).N) == 0) {
+			log.Info("    PublicKey modification")
+		}
+	}
+}
+
+func pemToCertificate(certPem string) (*x509.Certificate, error) {
+	block, _ := pem.Decode([]byte(certPem))
+	if block == nil {
+		return nil, errors.New("Fail decode pem")
+	}
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return nil, err
+	}
+	return cert, nil
+}
+
+func isCert(cert string) bool {
+	_, err := pemToCertificate(cert)
+	return err == nil
+}
+func check(cert string) bool {
+	rootPool := x509.NewCertPool()
+	certRoots, err := pemToCertificate(LetsencryptCert)
+	if err != nil {
+		log.Info("The root cert is not valid")
+	}
+	rootPool.AddCert(certRoots)
+	certCheck, err := pemToCertificate(cert)
+	if err != nil {
+		log.Info("The checked cert is not valid")
+	}
+	opts := x509.VerifyOptions{
+		DNSName: certCheck.DNSNames[0],
+		Roots:   rootPool,
+	}
+	if _, err := certCheck.Verify(opts); err != nil {
+		log.Info("certificate not valid: " + err.Error())
+		return false
+	}
+	log.Info("Cert OK")
+	return true
+
+}
+func renewCert(cert string) string {
+
+	oldcert, err := pemToCertificate(cert)
+	if err != nil {
+		log.Fatal(err)
+	}
+	//get the serial number and compute the URI
```

Comment:
Before comment add newline.

Reply:
Comments should be whole sentences, start with a capital letter and have a space between the `//` and the first letter.

Reply:
`// Get the serial number and compute the URI.`